src/Symfony/Component/Messenger/Tests/WorkerTest.php

```diff
@@ -313,12 +314,13 @@ public function testTimeoutIsConfigurable()
         $dispatcher = new EventDispatcher();
         $dispatcher->addSubscriber(new StopWorkerOnMessageLimitListener(5));
 
-        $worker = new Worker([$receiver], $bus, $dispatcher);
-        $startTime = microtime(true);
+        $clock = new MockClock();
+        $worker = new Worker([$receiver], $bus, $dispatcher, clock: $clock);
+        $startTime = $clock->now();
         // sleep .1 after each idle
         $worker->run(['sleep' => 100000]);
 
-        $duration = microtime(true) - $startTime;
+        $duration = $clock->now()->format('U.u') - $startTime->format('U.u');
```

Comment:
You could set the mock clock to a fixed date and time before constructing the worker. This way, you won't need to do this kind of calculation anymore, you could directly assert the state of `$clock->now()`.

Reply:
Indeed